aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -0,0 +1,183 @@
+/// This module supports functionality related to code management.
+module aptos_framework::code {
+    use std::string::String;
+    use std::error;
+    use std::signer;
+    use std::vector;
+
+    // ----------------------------------------------------------------------
+    // Code Publishing
+
+    /// The package registry at the given address.
+    struct PackageRegistry has key {
+        /// Packages installed at this address.
+        packages: vector<PackageMetadata>,
+    }
+
+    /// Metadata for a package.
+    struct PackageMetadata has store, copy, drop {
+        /// Name of this package.
+        name: String,
+        /// The upgrade policy of this package.
+        upgrade_policy: UpgradePolicy,
+        /// The list of modules installed by this package.
+        modules: vector<ModuleMetadata>,
+        /// Address aliases which where used when the modules above were compiled.
+        address_aliases: vector<AddressAlias>,
+        /// Dependencies which were used when this package was compiled.
+        deps: vector<PackageDep>,
+    }
+
+    /// Metadata about a module in a package.
+    struct ModuleMetadata has store, copy, drop {
```

Comment:
This looks expensive to store on chain. The source code can be large. I think the source code should be kept off chain, e.g. in the explorer system

Reply:
This was discussed with the storage guys and they seemed to be fine with it. In the bigger picture, source isn't that large.

We need this to compile against the package without having some other truth of source. 

We may later consider to put this into some kind of secondary blob store, but logically, it belongs to here.

Reply:
Makes sense. All these fields are optional though right?